TESTS/netsocket/dns/asynchronous_dns_simultaneous_cache.cpp

```diff
@@ -32,6 +32,7 @@ int result_exp_timeout;
 
 void ASYNCHRONOUS_DNS_SIMULTANEOUS_CACHE()
 {
+    nsapi_dns_reset();
```

Comment:
Looking at the comment below, this test relies on previous tests caching some entries.
When I ran it with these changes I got a failure on K64F+ESP8266... Perhaps we need to modify the expectations to:
```
    // 6th entry will return out of memory.
    TEST_ASSERT_EQUAL(MBED_CONF_APP_DNS_SIMULT_QUERIES, result_ok);
    TEST_ASSERT_EQUAL(1, result_no_mem);
```

Reply:
Changed as you suggested, could have been fixed also by doing 1 query beforehand to put it in cache and then this would have worked